### PR TITLE
Fixes Firefox WebSocket upgrade request in WebSocket engine

### PR DESCRIPTION
**Pull request by sab24**
This fixes issue [#2158 in czmq](https://github.com/zeromq/czmq/issues/2158) and [#18 in jszmq](https://github.com/zeromq/jszmq/issues/18) project

### CODE REVIEW

**Review by somdoron — CHANGES_REQUESTED** [review]
Can you fix the indentation of this line?

File: src/ws_engine.cpp [hunk 1]
```diff
@@ -453,9 +453,18 @@ bool zmq::ws_engine_t::server_handshake ()
...
-                    else if (strcasecmp ("connection", _header_name) == 0)
-                        _header_connection_upgrade =
-                          strcasecmp ("upgrade", _header_value) == 0;
+                    else if (strcasecmp ("connection", _header_name) == 0){
+			char *element = strtok (_header_value, ",");
+			while (element != NULL){
+			    while (*element == ' ')
+			        element++;
+                            if (strcasecmp ("upgrade", element) == 0){
+			        _header_connection_upgrade = true;
```
Comment (somdoron) [inline src/ws_engine.cpp:462]: Can you fix the indentation of this line?

**Comment by sab24** [comment]
Fixed the indentation issues, there were some problems with Vim

**Comment by somdoron** [comment]
@sab24 can you also include a relicense grant? 

https://github.com/zeromq/libzmq/tree/master/RELICENSE

**Pull request merged**